src/components/Form/ProfileForm.tsx

```diff
@@ -80,17 +95,29 @@ function ProfileForm({ callback }: { callback?: () => void }) {
           name: values.name,
           about: values.about,
           skills: values.skills,
-          web3mailPreferences: user.description?.web3mailPreferences,
+          // Don't update this field if User is Collaborator => Needs to be Owner
+          web3mailPreferences:
+            isBuilderPlaceCollaborator && !isBuilderPlaceOwner
+              ? ''
+              : user.description?.web3mailPreferences,
         };
 
         let cid = await talentLayerClient.profile.upload(profile);
 
         let tx;
         if (isActiveDelegate) {
-          const response = await delegateUpdateProfileData(chainId, user.id, user.address, cid);
+          const response = await delegateUpdateProfileData(
+            chainId,
+            isBuilderPlaceCollaborator ? builderPlace.ownerTalentLayerId : user.id,
```

Comment:
i guess we will have to do this check a lot of time, so it is better to have this logic directly in the context